Gems/LandscapeCanvas/Code/Source/LandscapeCanvasSystemComponent.cpp

```diff
@@ -82,29 +79,58 @@
 
 namespace LandscapeCanvas
 {
+    namespace Internal
+    {
+        // The FastNoise gem is optional, so we need to keep track of its component type ID
+        // ourselves since we can't rely on the headers being there.
+        static constexpr const char* EditorFastNoiseGradientComponentTypeId = "{FD018DE5-5EB4-4219-9D0C-CB3C55DE656B}";
+
+        // The Vegetation gem is optional, so we need to keep track of the component type IDs
+        // ourselves since we can't rely on the headers being there.
+        namespace Vegetation
+        {
+            static constexpr const char* EditorAreaBlenderComponentTypeId = "{374A5C69-A252-4C4B-AE10-A673EF7AFE82}";
+            static constexpr const char* EditorBlockerComponentTypeId = "{9E765835-9CEB-4AEC-A913-787D3D21451D}";
+            static constexpr const char* EditorMeshBlockerComponentTypeId = "{130F5DFF-EF6F-4B37-8717-194876DE12DB}";
+            static constexpr const char* EditorSpawnerComponentTypeId = "{DD96FD51-A86B-48BC-A6AB-89183B538269}";
+            static constexpr const char* EditorDistanceBetweenFilterComponentTypeId = "{78DE1245-7023-40D6-B365-CC45EB4CE622}";
+            static constexpr const char* EditorDistributionFilterComponentTypeId = "{8EDD1DA2-B597-4BCE-9285-C68886504EC7}";
```

Comment:
Just thinking out loud...

At some point we should consider redesigning how nodes work if possible, so that we can define the nodes themselves in the gems that they're associated with. We could then find them all dynamically at runtime through the serialization context (Descriptor::RefreshSpawnerTypeList does something like this in Vegetation).

It's a pretty big change and would require having a common node base class available to all these gems though.